### ISSUE TITLE
WriteFreely won't start on older OSes due to glibc issue

### ISSUE THREAD

**Issue by thebaer** [writefreely/writefreely#1064]
### Describe the bug

When attempting to run the WriteFreely v0.15.1 release build on anything before Ubuntu 20.04, it fails to start with this error message:

```
./writefreely: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by ./writefreely)
```

This didn't happen with Go 1.19, but with the update to Go 1.21 (#1025), we now see this on systems that don't have this version of `glibc` (see golang/go#58550). We can't use the `CGO_ENABLED=0` fix, because we need CGO for sqlite.

Instead, we should statically compile WriteFreely for release -- see https://github.com/golang/go/issues/26492#issuecomment-435462350 for a potential starting point.

### Workaround

Admins can compile WriteFreely from scratch on their target server by cloning the repository and running:

```
make build
```

### Application configuration
 - **Single mode or Multi-user mode?** n/a
 - **Database?** n/a
 - **Open registration?** n/a
 - **Federation enabled?** n/a

**Version or last commit**: v0.15.1 (release build)

**Comment by sahilmulla** [comment]
@thebaer can I work on this?

**Comment by thebaer** [comment]
@sahilmulla Yes, please do!

**Comment by sahilmulla** [comment]
@thebaer 

- runs on ```ubuntu:18.04``` ```ldd:2.27``` with ```CGO_ENABLED=1``` by adding ```-ldflags="-extldflags=-static"```
binary size is **43M**

any other platforms need to be tested?

**Comment by thebaer** [comment]
Thanks! We want to be sure it works on Windows, Mac, and Linux. See the `build-*` directives in [Makefile](https://github.com/writefreely/writefreely/blob/develop/Makefile) for all platforms / architectures.

Do you mind opening a PR for this once you've tested? Sounds like you'll just need to change the `LDFLAGS` variable in that file.

**Comment by sahilmulla** [comment]
@thebaer ok will test on windows and mac and make a PR

**Comment by sahilmulla** [comment]
@thebaer just curious, how you approach testing on multiple platforms? 

I got a Windows machine, I used ubuntu containers to build and test for Linux.

Also for make build tasks can we make them platform agnostic? We could try moving them to go-task?

**Comment by thebaer** [comment]
For me it's just a manual process. But I'm open to anything we could set something up to do that in a more automated way!

We could also try moving to go-task, yep. If you want to tackle that please feel free, just be sure to separate that into a different PR so we can address each issue separately.

**Comment by ellotheth** [comment]
Not sure if this is appropriate here, but calling it out since I haven't seen it mentioned: I've been running WriteFreely on Alpine with the `gcompat` package for glibc compatibility, and I'm getting the same glibc-related errors (in my case, `fcntl64: symbol not found`). I've tried several combinations of Alpine versions 3.18-3.20 and `edge`, with the `testing` package repo.

**Comment by DonPiotr** [comment]
> Not sure if this is appropriate here, but calling it out since I haven't seen it mentioned: I've been running WriteFreely on Alpine with the `gcompat` package for glibc compatibility, and I'm getting the same glibc-related errors (in my case, `fcntl64: symbol not found`). I've tried several combinations of Alpine versions 3.18-3.20 and `edge`, with the `testing` package repo.

Same here

Without gcompat I get:
`-bash: ./writefreely: cannot execute: required file not found`

**Comment by sahilmulla** [comment]
@thebaer will look into go-task after this

**Comment by thebaer** [comment]
@ellotheth @DonPiotr Are you able to compile the binary and see if the fix in #1116 works for you?

**Comment by DonPiotr** [comment]
> @ellotheth @DonPiotr Are you able to compile the binary and see if the fix in #1116 works for you?

Yes, it is compiling and working.

**Comment by helen001122** [comment]
[Wordle Unlimited](https://wordleunlimited.com/) is completely free to play. You don’t need to register, download an app, or pay any fees. The game works entirely within your browser and offers an ad-light experience compared to many other online games.
